astroid/inference.py

```diff
@@ -100,7 +106,10 @@ def _infer_sequence_helper(node, context=None):
 
 
 @decorators.raise_if_nothing_inferred
-def infer_sequence(self, context=None):
+def infer_sequence(
+    self: Union[nodes.List, nodes.Tuple, nodes.Set],
+    context: Optional[InferenceContext] = None,
+) -> Iterator[Union[nodes.List, nodes.Tuple, nodes.Set]]:
```

Comment:
Isn't the input `self` type also the same as the return type?
I would suggest a TypeVar bound to `BaseContainer` here.